src/app/plugins/animal.plugin.ts

```diff
@@ -0,0 +1,181 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
+import ms from 'ms';
+
+export default class AnimalPlugin extends Plugin {
+  public commandName: string = 'animal';
+  public name: string = 'Animal Plugin';
+  public description: string = 'Get wild with a fraction of gods creatures';
+  public usage: string =
+    'animal <Species (optional)> <subSpecies(Optional)> | animal list (species or <species> or subSpecies)';
+  public override pluginAlias = ['wild', 'species'];
+  public permission: ChannelType = ChannelType.Public;
+  public override pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://api.fetchit.dev/';
+
+  private _validAnimals: Set<string> = new Set([]); // All animals and species
+  private _animals: ISubSpecies[] = []; // All species.
+
+  private _ANIMAL_UPD_THRESH: number = ms('1d');
+  private _LAST_UPD_TIME: number = 0;
+
+  constructor(public container: IContainer) {
+    super();
+    this._updateAnimalAPIData();
+  }
+
+  public execute(message: IMessage, args?: string[]) {
+    this._updateAnimalAPIData();
+    const input = args?.length ? args : [this._getRandomAnimal()];
+
+    // Checks if they want to know what species/ subspecies are available.
+    if (input[0].startsWith('list')) {
+      this._pickListType(message, input[1]);
+      return;
+    }
+
+    // Preps the lookup data, and checks if it's valid.
+    const animalLookUp: string = input.join('_');
+    if (!this._validAnimals.has(animalLookUp)) {
+      message.reply(`\`${input}\`is a invalid species/ sub-species.`);
+      return;
+    }
+
+    let searchTerms: string = `name=${animalLookUp}`;
+    const animalIndex = this._animals.findIndex((animal) => animal.species === animalLookUp);
+    const species = this._findAnimalsFromSubspecies(animalLookUp)?.species;
+
+    // Checks if it's a sub-species, if so it retrieves its species.
+    if (species !== undefined) {
+      searchTerms = `name=${species}&sub-species=${animalLookUp}`;
+    } else if (this._animals[animalIndex].subSpecies.length > 0) {
+      // Detects if the species has subspecies.  Then applies a random sub-species accordingly.
+      const subSpeciesList: string[] = this._animals[animalIndex].subSpecies;
+      const randomSubspecies = subSpeciesList[Math.floor(Math.random() * subSpeciesList.length)];
+
+      searchTerms = `name=${animalLookUp}&sub-species=${randomSubspecies}`;
```

Comment:
instead of having this pattern in multiple places, it could be cool to make a function that builds that url for us
```ts
const createSearchTerm = (species: string, sub?: string) => {
    let searchTerm = `name=${species}`;
    if(sub){
        searchTerm += `&sub-species=${sub}`
    }

    return searchTerm;
}
```